invalid memory address or nil pointer dereference

just now installed steampipe and vault plugin. every command related to vault fails with the same error
` select * from vault_engine
Error: rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
> select * from vault_engine;
Error: rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
> select * from vault.vault_engine;
Error: rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
> select * from vault.vault_kv_secret
Error: rpc error: code = Internal desc = list call listSecrets failed with panic runtime error: invalid memory address or nil pointer dereference
> select * from vault.vault_sys_health
Error: rpc error: code = Internal desc = list call getSysHealth failed with panic runtime error: invalid memory address or nil pointer dereference
`
VAULT_TOKEN and VAULT_ADDR are set

other plugins such as gitlab are working fine
`steampipe plugin list`

+----------------------------------------------------+---------+-------------+
| Name                                               | Version | Connections |
+----------------------------------------------------+---------+-------------+
| hub.steampipe.io/plugins/theapsgroup/gitlab@latest | 0.0.3   | gitlab      |
| hub.steampipe.io/plugins/theapsgroup/vault@latest  | 0.0.4   | vault       |
| hub.steampipe.io/plugins/turbot/aws@latest         | 0.34.0  | aws         |
| hub.steampipe.io/plugins/turbot/docker@latest      | 0.0.2   | docker      |
| hub.steampipe.io/plugins/turbot/steampipe@latest   | 0.1.3   | steampipe   |
+----------------------------------------------------+---------+-------------+

@e-gineer :+1: log content related to the error
`021-10-05 08:48:29.938 GMT [49229] LOG:  statement: BEGIN READ WRITE
2021-10-05 08:48:29.938 GMT [49229] LOG:  statement: select * from vault_engine
2021/10/05 10:48:29 [INFO] 
******************************************************

                steampipe postgres fdw init

******************************************************
2021/10/05 10:48:29 [INFO] Log level WARN
2021-10-05T10:48:29.971+0200 [ERROR] plugin.steampipe-plugin-vault.plugin: [ERROR] streamRows err chan select: rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05T10:48:29.971+0200 [WARN]  hub: stream receive error rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference (0xc00030ae70)
2021-10-05 08:48:29.971 GMT [49229] ERROR:  rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05 08:48:29.971 GMT [49229] STATEMENT:  select * from vault_engine
2021-10-05 08:48:29.972 GMT [49229] LOG:  statement: ROLLBACK
2021-10-05 08:48:38.550 GMT [49229] LOG:  statement: BEGIN READ WRITE
2021-10-05 08:48:38.550 GMT [49229] LOG:  statement: select * from vault_engine;
2021-10-05T10:48:38.564+0200 [ERROR] plugin.steampipe-plugin-vault.plugin: [ERROR] streamRows err chan select: rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05T10:48:38.564+0200 [WARN]  hub: stream receive error rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference (0xc0000f80b0)
2021-10-05 08:48:38.564 GMT [49229] ERROR:  rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05 08:48:38.564 GMT [49229] STATEMENT:  select * from vault_engine;
2021-10-05 08:48:38.565 GMT [49229] LOG:  statement: ROLLBACK
2021-10-05 08:48:51.514 GMT [49229] LOG:  statement: BEGIN READ WRITE
2021-10-05 08:48:51.514 GMT [49229] LOG:  statement: select * from vault.vault_engine;
2021-10-05T10:48:51.521+0200 [ERROR] plugin.steampipe-plugin-vault.plugin: [ERROR] streamRows err chan select: rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05T10:48:51.521+0200 [WARN]  hub: stream receive error rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference (0xc000219810)
2021-10-05 08:48:51.522 GMT [49229] ERROR:  rpc error: code = Internal desc = list call listEngines failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05 08:48:51.522 GMT [49229] STATEMENT:  select * from vault.vault_engine;
2021-10-05 08:48:51.523 GMT [49229] LOG:  statement: ROLLBACK
2021-10-05 08:49:20.601 GMT [49229] LOG:  statement: BEGIN READ WRITE
2021-10-05 08:49:20.602 GMT [49229] LOG:  statement: select * from vault.vault_kv_secret
2021-10-05T10:49:20.608+0200 [ERROR] plugin.steampipe-plugin-vault.plugin: [ERROR] streamRows err chan select: rpc error: code = Internal desc = list call listSecrets failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05T10:49:20.608+0200 [WARN]  hub: stream receive error rpc error: code = Internal desc = list call listSecrets failed with panic runtime error: invalid memory address or nil pointer dereference (0xc0002198c0)
2021-10-05 08:49:20.608 GMT [49229] ERROR:  rpc error: code = Internal desc = list call listSecrets failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05 08:49:20.608 GMT [49229] STATEMENT:  select * from vault.vault_kv_secret
2021-10-05 08:49:20.609 GMT [49229] LOG:  statement: ROLLBACK
2021-10-05 08:49:54.606 GMT [49229] LOG:  statement: BEGIN READ WRITE
2021-10-05 08:49:54.607 GMT [49229] LOG:  statement: select * from vault.vault_sys_health
2021-10-05T10:49:54.612+0200 [ERROR] plugin.steampipe-plugin-vault.plugin: [ERROR] streamRows err chan select: rpc error: code = Internal desc = list call getSysHealth failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05T10:49:54.612+0200 [WARN]  hub: stream receive error rpc error: code = Internal desc = list call getSysHealth failed with panic runtime error: invalid memory address or nil pointer dereference (0xc000219970)
2021-10-05 08:49:54.612 GMT [49229] ERROR:  rpc error: code = Internal desc = list call getSysHealth failed with panic runtime error: invalid memory address or nil pointer dereference
2021-10-05 08:49:54.612 GMT [49229] STATEMENT:  select * from vault.vault_sys_health
2021-10-05 08:49:54.613 GMT [49229] LOG:  statement: ROLLBACK
2021-10-05 08:50:45.109 GMT [49760] LOG:  statement: ;
`
@Bonemind : vault cli is working fine tested with "vault kv get", config file is default (nothing set, use VAULT_ADDR and VAULT_TOKEN env variable

system is kubuntu 20.04 latest updates included. tested with vault version 1.8.3

--- a/vault/utils.go
+++ b/vault/utils.go
@@ -17,6 +17,7 @@ import (
 func connect(ctx context.Context, d *plugin.QueryData) (*api.Client, error) {
 	addr := os.Getenv("VAULT_ADDR")
 	tkn := os.Getenv("VAULT_TOKEN")
+	defaultAuthType := "token"
 
 	// In line with the vault CLI, these values can be set through environment variables.
 	vaultConfig := GetConfig(d.Connection)
@@ -27,6 +28,10 @@ func connect(ctx context.Context, d *plugin.QueryData) (*api.Client, error) {
 
 	if vaultConfig.Token == nil {
 		vaultConfig.Token = &tkn
+	}
+
+	if vaultConfig.AuthType == nil {
+		vaultConfig.AuthType = &defaultAuthType
 	}
 
 	if *vaultConfig.Address == "" {

